deno/src/platform.ts

```diff
@@ -46,7 +48,11 @@ export const inputFileData = Symbol('InputFile data')
  * Reference](https://core.telegram.org/bots/api#inputfile).
  */
 export class InputFile {
-    public readonly [inputFileData]: ConstructorParameters<typeof InputFile>[0]
```

Comment:
For 2.0, `InputFile` should accept `Blob` instead of `path: string` (https://github.com/nodejs/node/issues/37340), and expose just `name: string` and a `.stream()` method.

Reply:
That sounds like a good idea, that would simply the abstraction 👍🏻

Reply:
Further, you could turn `InputFile` into an interface, and provide `class InputFileAsyncIterable` instead, perhaps some others. Zero type-switching in your code!

You could still easily detect `InputFile`s with

```ts
function isInputFile(value: any): value is InputFile {
  return value != null && typeof value.stream === 'function'
}
```

Note that web `File` already satisfies

```ts
interface InputFile {
  readonly name: string
  readonly stream: () => ReadableStream
}
```

Reply:
Again, the goal is to be explicit about time uploads by constructing an `InputFile` object. When bot code is read, it should be immediately obvious that a file upload will be performed. Permitting lots of things in the API directly contradicts this because it allows to circumvent the `InputFile` constructor.

That being said, I'm open to change completely how `InputFile` and client work together if there is a better solution.

Reply:
I'd argue that `new upload.Buffer(uint8array, name)` or `new upload.Iterable(streamOrIterable, name)` or `await Deno.getFile(path)` is more explicit than `new InputFile(oneOfFourTypes)`.

I don't think it makes sense to ever merge this PR. Better open a fresh one when working on 2.0.

Reply:
On 2nd thought, this could be made mergable:

1. Switch to `deno2node`,
2. Shim `Blob`,
3. Move `Blob` detection to `Client`.

But there doesn't seem to be any demand.